src/main/java/domain/JanggiRunner.java

```diff
@@ -0,0 +1,93 @@
+package domain;
+
+import domain.piece.Piece;
+import domain.piece.PieceFactory;
+import domain.piece.strategy.HorseElephantSetupStrategy;
+import java.util.List;
+import view.InputView;
+import view.OutputView;
+
+public class JanggiRunner {
+    private final InputView inputView;
+    private final OutputView outputView;
+
+    public JanggiRunner(InputView inputView, OutputView outputView) {
+        this.inputView = inputView;
+        this.outputView = outputView;
+    }
+
+    public void run() {
+        JanggiGame janggiGame = initializeGame();
+        showInitializedBoardResult(janggiGame);
+        startGame(janggiGame);
+    }
+
+    private void showInitializedBoardResult(JanggiGame janggiGame) {
+        List<Piece> alivePieces = janggiGame.getAlivePieces();
+        outputView.printBoard(alivePieces);
+    }
+
+    private void startGame(JanggiGame janggiGame) {
+        executeGame(janggiGame);
+        showWinner(janggiGame);
+    }
+
+    private void executeGame(JanggiGame janggiGame) {
+        TeamType nowTurn = TeamType.CHO;
+        while (isGameInProgress(janggiGame)) {
```

Comment:
`isGameInProgress` 메소드를 `JanggiRunner` 에서 추가로 만들어서 호출하지 말고
`JanggiGame` 내부에서 만든 메소드를 바로 호출하면 어떨까요?
불필요한 `depth`는 줄이는 것이 좋습니다.


```suggestion
        while (janggiGame.isGameInProgress()) {
```

Reply:
조건에 대한 부정문을 제거하기 위해 `JanggiRunner` 내부에 `IsGameInProgress()` 메서드를 두었는데 오히려 불필요한 `depth`가 생긴 것 같아요. `JanggiGame` 내부에서 `isInProgress()`메서드를 호출하도록 변경하겠습니다!